### DISCUSSION TITLE
PSBT field names

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1054] [Q&A] [answered]
Two questions relating to PSBT field names, please:

1. Why do we not use the matching identifiers for _all_ fields as used it BIP-174? For example, some are the same `PSBT_IN_TAP_INTERNAL_KEY ` => `tap_internal_key` but some are different `PSBT_IN_TAP_BIP32_DERIVATION` => `tap_key_origins`?

2. Why are some fields missing? For example, `PSBT_IN_REQUIRED_TIME_LOCKTIME`?

**Comment by DanGould** [comment, marked as answer]
1. naming is hard. i think consistency is a good idea

2. I think the missing fields are new to BIP370 psbt v2 and just haven't been added yet. the BIP174 spec was updated to include the new v2 fields

**Reply by apoelstra** [reply]
I think both these points are correct.

**Comment by sanket1729** [comment]
@tcharding, I copied the names from BIP371 in the bitcoin core pull requests. https://github.com/bitcoin/bitcoin/pull/22558. I don't oppose renaming the fields to match the spec.
